gmso/external/convert_parmed.py

```diff
@@ -339,146 +386,296 @@ def _angle_types_from_pmd(structure, angle_types_member_map=None):
         corresponding GMSO.AngleType object.
     """
     pmd_top_angletypes = dict()
-    angle_types_member_map = _assert_dict(
-        angle_types_member_map, "angle_types_member_map"
+    harmonicbond_potential = lib["HarmonicAnglePotential"]
+    name = harmonicbond_potential.name
+    expression = harmonicbond_potential.expression
+    variables = harmonicbond_potential.independent_variables
+
+    if not structure.angle_types:
+        return pmd_top_angletypes
+    unique_angle_types = list(
+        dict.fromkeys(
+            [
+                (
+                    angle.type.k,
+                    angle.type.theteq,
+                    (
+                        angle.atom2.type,
+                        *sorted((angle.atom1.type, angle.atom3.type)),
+                    ),
+                )
+                for angle in structure.angles
+            ]
+        )
     )
-
-    for angletype in structure.angle_types:
+    for angletype in unique_angle_types:
         angle_params = {
-            "k": (2 * angletype.k * u.Unit("kcal / (rad**2 * mol)")),
-            "theta_eq": (angletype.theteq * u.degree),
+            "k": (2 * angletype[0] * u.Unit("kcal / (rad**2 * mol)")),
+            "theta_eq": (angletype[1] * u.degree),
         }
-        expr = gmso.AngleType._default_potential_expr()
-        expr.parameters = angle_params
-        # Do we need to worry about Urey Bradley terms
+        # TODO: we need to worry about Urey Bradley terms
         # For Urey Bradley:
         # k in (kcal/(angstrom**2 * mol))
         # r_eq in angstrom
-        member_types = angle_types_member_map.get(id(angletype))
         top_angletype = gmso.AngleType(
-            potential_expression=expr, member_types=member_types
+            name=name,
+            parameters=angle_params,
+            expression=expression,
+            independent_variables=variables,
+            member_types=angletype[2],
```

Comment:
Based on how atoms order defined above, wouldn't the `angletype[2]` have the order of `(middle-point, edge-point-1, edge-point-2)`, should that be reordered here to conform with the angle convention where the middle atom(type) is in the middle?

Reply:
Yep that's right. In GMSO/Foyer we define it as the central atom in the middle. I'm guessing that no error arises here because the angle_type is is still associated with the angle (edge-point-1, middle-point, edge-point-2). So the angle itself is defined correctly, and if you printed out:
```python
# case 1: angle with reversed member_types
print(list(map(lambda x: x.atom_type.name, angle.connection_members)) # -> [A, B, C]
print(angle.angle_type.member_types) # -> [B, A, C]

# case 2: for some angle_mirror that is a mirror to angle
print(list(map(lambda x: x.atom_type.name, angle_mirror.connection_members)) # -> [C, B, A]
print(angle_mirror.angle_type.member_types) # -> [B, A, C]

# case 3: for some angle and sorts member types by GMSO addition
print(list(map(lambda x: x.atom_type.name, angle_gmso.connection_members)) # -> [A, B, C]
print(angle_mirror.angle_type.member_types) # -> [C, B, A]
```

So I guess it just depends on what we exactly what member types to entail. I think the central atom should still always go in the middle. However, I think the outer and inner atoms should reflect the sorted ordering of the type, not the actual ordering if the atoms in angle.connection_members:

Fixed code to generate this behavior:
```python
        top_angletype = gmso.AngleType(
            name=name,
            parameters=angle_params,
            expression=expression,
            independent_variables=variables,
            member_types=[angle_type[2][i] for i in [1,0,2]
        )

Reply:
Based on conversation with @daico007, we want to sort all member_types by the connection.connection_member indices. In this case, types can be made that are mirrors of each other, and only filtered into identical types if the correct filter is passed to the PotentialViews object. This change will be reflected shortly and applied to all connections in the parmed structure.

Reply:
Addressed in [408ef1b](https://github.com/mosdef-hub/gmso/pull/721/commits/408ef1ba5b6b613ad3f9f541d395a4a99daaa1e5)